gui/src/app/pages/HomePage/LoadProjectWindow.tsx

```diff
@@ -89,58 +81,85 @@ const LoadProjectWindow: FunctionComponent<LoadProjectWindowProps> = ({
     [filesUploaded, onClose, update],
   );
 
-  useEffect(() => {
-    if (!filesUploaded) return;
-    if (filesUploaded.length === 1 && !filesUploaded[0].name.endsWith(".zip")) {
-      // The user has uploaded a single file and it is not a zip file. In
-      // this case we want to give the user the option whether or not to
-      // replace the current project.
-      setShowReplaceProjectOptions(true);
-    } else {
-      // Otherwise, we just go ahead and import the files, replacing the
-      // entire project
-      importUploadedFiles({ replaceProject: true });
-    }
-  }, [filesUploaded, importUploadedFiles]);
+  const onUpload = useCallback(
+    (fs: { name: string; content: ArrayBuffer }[]) => {
+      if (fs.length === 1 && fs[0].name.endsWith(".zip")) {
+        setFilesUploaded(fs);
+        importUploadedFiles({ replaceProject: true });
```

Comment:
This is not going to work. The setFilesUploaded(fs) is not going to take effect until the next render, so importUploadedFiles() should not be called immediately.

What I would suggest  is to have a separate useEffect that checks for the case of a single .zip file and then calls the importFilesUploaded()

Reply:
Yep, should have tested this! I ended up splitting up the zip logic from the function that gets called when the buttons are clicked